agent/grpc-external/server.go

```diff
@@ -80,3 +96,24 @@ func NewServer(
 	}
 	return grpc.NewServer(opts...)
 }
+
+func makeDirector(serverConn *grpc.ClientConn, logger hclog.Logger) func(ctx context.Context, fullMethodName string) (context.Context, *grpc.ClientConn, error) {
+	return func(ctx context.Context, fullMethodName string) (context.Context, *grpc.ClientConn, error) {
+		var mdCopy metadata.MD
+		md, ok := metadata.FromIncomingContext(ctx)
+		if !ok {
+			mdCopy = metadata.MD{}
+		} else {
+			mdCopy = md.Copy()
+		}
+		outCtx := metadata.NewOutgoingContext(ctx, mdCopy)
+
+		logger.Info("forwarding the request to the consul server", "method", fullMethodName)
+		// throw unimplemented error if the method is not meant to be forwarded
+		if !strings.Contains(fullMethodName, FORWARD_SERVICE_NAME_PREFIX) {
```

Comment:
Nit. Maybe we should use `HasPrefix()` here, instead of `Contains()`?

Reply:
Sorry about that, I should have updated this API before. Updated